sdks/java/core/src/test/java/org/apache/beam/sdk/io/WriteTest.java

```diff
@@ -203,13 +269,20 @@ private static void runWrite(
     }
 
     TestSink sink = new TestSink();
+    Write.Bound<String> write = Write.to(sink);
+    if (numConfiguredShards != null) {
+      write = write.withNumShards(numConfiguredShards);
+    }
     p.apply(Create.timestamped(inputs, timestamps).withCoder(StringUtf8Coder.of()))
      .apply(transform)
-     .apply(Write.to(sink));
+     .apply(write);
 
     p.run();
     assertThat(sinkContents, containsInAnyOrder(inputs.toArray()));
```

Comment:
Balancing the output across multiple sinks is important so can we test that somehow?